### PR TITLE
Update certificate-credentials.md

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: lib/msal-node/docs/certificate-credentials.md [hunk 1]
```diff
@@ -2,59 +2,25 @@
...
+* `managed identity`: this is a certificatless scenario, where trust is established via the Azure infrastructure. No secret / certificate management is required. See https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/
+* `clientSecret`: a secret string generated during the app registration, or updated post registration for an existing application. This is not recommended for production.
+* `clientCertificate`: a certificate set during the app registration, or updated post registration for an existing application. The certificate needs to have the private key, because it will be used for signing [an assertion](https://learn.microsoft.com/azure/active-directory/develop/certificate-credentials) that MSAL generates.
+* `clientAssertion`: instead of letting MSAL create an [assertion](https://learn.microsoft.com/azure/active-directory/develop/certificate-credentials), the app developer takes control. Useful for adding extra claims to the assertion or for using KeyVault for signing, instead of a local certificate.
```
Comment (sameerag) [inline lib/msal-node/docs/certificate-credentials.md]: Don't we need to define what `clientAssertion` is? The above implies that difference between `clientCertificate` and `clientAssertion` is only the generation/ownership?

File: lib/msal-node/docs/certificate-credentials.md [hunk 2]
```diff
@@ -2,59 +2,25 @@
...
-> ```powershell
->$cert=New-SelfSignedCertificate -Subject "CN=example.com" -CertStoreLocation "Cert:\CurrentUser\My"  -KeyExportPolicy Exportable -KeySpec Signature
->```
->
-> This command will generate two files: *example.cer* (public key) and *example.pfx* (public key + encrypted private key, usually formatted in *PKCS#12*). For more information, see: [Create a self-signed public certificate to authenticate your application](https://docs.microsoft.com/azure/active-directory/develop/howto-create-self-signed-certificate)
-
-> :information_source: Certificate files come in various file extensions, such as *.crt*, *.csr*, *.cer*, *.pem*, *.pfx*, *.key*. For file type conversions, you can use *OpenSSL*. See below for [pfx to pem conversion](#optional-converting-pfx-to-pem). For other type of conversions, please refer to: [SSL/TLS Certificate File Types/Extensions](https://docs.microsoft.com/archive/blogs/kaushal/various-ssltls-certificate-file-typesextensions).
-
-### Registering self-signed certificates
+You can create a self-signed certificate using PowerShell or using Azure KeyVault.
```
Comment (sameerag) [inline lib/msal-node/docs/certificate-credentials.md]: Do we need links/docs here?

File: lib/msal-node/docs/certificate-credentials.md (hunk 1 above)
Comment (bgavrilMS) [reply lib/msal-node/docs/certificate-credentials.md]: Thx, I tried to clarify that a cert is still needed.

File: lib/msal-node/docs/certificate-credentials.md (hunk 2 above)
Comment (bgavrilMS) [reply lib/msal-node/docs/certificate-credentials.md]: I added a link to PowerShell command. The link around cert format is an internal blog and in my experience they tend to go 404. In any case, it's not important info.

**Review by sameerag — APPROVED** [review]

File: lib/msal-node/docs/certificate-credentials.md [hunk 3]
```diff
@@ -2,59 +2,25 @@
 
 > :warning: Before you start here, make sure you understand [Initialize confidential client applications](./initialize-confidential-client-application.md).
 
-You can build confidential client applications with MSAL Node (web apps, daemon apps etc). A **client credential** is mandatory for confidential clients. Client credential can be a:
+You can build confidential client applications with MSAL Node (web apps, daemon apps etc). A **client credential** is mandatory for confidential clients. Client credential can be:
 
-* `clientSecret`: a secret string generated during the app registration, or updated post registration for an existing application.
-* `clientCertificate`: a certificate set during the app registration, or updated post registration for an existing application. The `thumbprint` is a *X.509 SHA-1* thumbprint of the certificate, and the `privateKey` is the PEM encoded private key. `x5c` is the optional *X.509* certificate chain used in [subject name/issuer auth scenarios](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/sni.md).
```
Comment (Robbie-Microsoft) [inline lib/msal-node/docs/certificate-credentials.md:8]: Why did you delete the X.509 info?

File: lib/msal-node/docs/certificate-credentials.md [hunk 4]
```diff
@@ -2,59 +2,25 @@
...
 
-## Using certificates
+Note: 1p apps may be required to also send `x5c`. This is the *X.509* certificate chain used in [subject name/issuer auth scenarios](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-node/docs/sni.md).
 
-This section covers creating a self-signed certificate and initializing a confidential client. For an implementation, see the code sample: [auth-code-with-certs](../../../samples/msal-node-samples/auth-code-with-certs)
+## Using certificates securely
 
-### Generating self-signed certificates
+Certificates should be stored and deployed safely. Consider using Azure KeyVault for certificate storage and rotation. Passwords should never be hardcoded in source code. 
+Please see https://learn.microsoft.com/azure/active-directory/develop/security-best-practices-for-app-registration#certificates-and-secrets for more information
```
Comment (Robbie-Microsoft) [inline lib/msal-node/docs/certificate-credentials.md]: nit: "see [certificates and secrets](url) for more information"

**Review by Robbie-Microsoft — APPROVED** [review]

File: lib/msal-node/docs/certificate-credentials.md (hunk 4 above)
Comment (localden) [reply lib/msal-node/docs/certificate-credentials.md]: +1 - let's make sure that we don't include raw URLs like this @bgavrilMS

File: lib/msal-node/docs/certificate-credentials.md (hunk 3 above)
Comment (bgavrilMS) [reply lib/msal-node/docs/certificate-credentials.md:8]: Right, adding it back.
